pkg/reconciler/ingress/ingress.go

```diff
@@ -261,11 +260,10 @@ func (r *Reconciler) reconcileVirtualServices(ctx context.Context, ia *v1alpha1.
 		}
 		kept.Insert(d.Name)
 	}
+
 	// Now, remove the extra ones.
 	vses, err := r.virtualServiceLister.VirtualServices(resources.VirtualServiceNamespace(ia)).List(
-		labels.Set(map[string]string{
-			serving.RouteLabelKey:          ia.GetLabels()[serving.RouteLabelKey],
-			serving.RouteNamespaceLabelKey: ia.GetLabels()[serving.RouteNamespaceLabelKey]}).AsSelector())
+		labels.Set(map[string]string{networking.IngressLabelKey: ia.Name}).AsSelector())
```

Comment:
I don't think this rolls out cleanly.  I think this could orphan VirtualServices created by the earlier version.

Reply:
Done.